plugins/alexa/src/main.ts

```diff
@@ -190,18 +214,41 @@ class AlexaPlugin extends AutoenableMixinProvider implements HttpRequestHandler,
             body.grant_type = 'refresh_token';
         }
 
+        const self = this;
+
         const accessTokenPromise = (async () => {
             const response = await axios.post('https://api.amazon.com/auth/o2/token', new URLSearchParams(body).toString(), {
                 headers: {
                     'Content-Type': 'application/x-www-form-urlencoded',
                 }
+            }).catch(error => {
+                switch (error?.response?.data?.error) {
+                    case 'invalid_client':
+                    case 'invalid_grant':
+                    case 'unauthorized_client':
+                        self.storageSettings.values.tokenInfo = undefined;
+                        self.log.e(error?.response?.data?.error_description);
+                        break;
+
+                    case 'authorization_pending':
+                        self.log.w(error?.response?.data?.error_description);
+                        break;
+                    
+                    case 'expired_token':
+                        self.accessToken = undefined;
+                        break;
+
+                    default:
+                        self.console.error(error?.response?.data);
+                }
```

Comment:
modified this change to rethrow the error here. otherwise it returns an undefined response which then fails obscurely.

Reply:
I was going to let these get swallowed, as they are non-actionable by us at the moment.